app/Providers/NovaServiceProvider.php

```diff
@@ -61,7 +63,13 @@ protected function routes(): void
     protected function gate(): void
     {
         Gate::define('viewNova', static function (User $user): bool {
-            return $user->can('access-nova');
+            return Cache::remember(
```

Comment:
Will Nova access changes still take effect immediately?

Reply:
No, this would only get flushed during deployments.

Reply:
I guess it's just something to keep in mind when giving new people access to Nova for the first time. But it sounds like that should be a case where the cache key for a user should be flushed, if that's a thing.

Reply:
I don't think there's a good way to do that right now.

Reply:
It would take a bit of effort but a listener for role assignment for the common roles that grant Nova access would probably do it. I would at least make an issue for that.

Reply:
Opened #2497